### PR TITLE
feat: extract metrics from overview app

### PR DESCRIPTION
**Pull request by YiniXu9506**
- Make the overview page is consistent with the one online.
- Build a new app called `Metrics` to show `TiDB performance overview` metrics.

### CODE REVIEW

File: ui/packages/tidb-dashboard-for-op/src/apps/Metrics/context.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  getAlertManagerCounts(address: string, options?: ReqConfig) {
+    return client.getInstance().getAlertManagerCounts({ address }, options)
+  }
+}
+
+const ds = new DataSource()
+
+export const ctx: IMetricsContext = {
+  ds,
+  cfg: { apiPathBase: client.getBasePath() }
```
Comment (shhdgit) [inline ui/packages/tidb-dashboard-for-op/src/apps/Metrics/context.ts:57]: Why we need apiPathBase here?

File: ui/packages/tidb-dashboard-for-op/src/apps/Metrics/context.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    return client.getInstance().getTiDBTopology(options)
+  }
+
+  getStoreTopology(options?: ReqConfig) {
+    return client.getInstance().getStoreTopology(options)
+  }
+
+  getPDTopology(options?: ReqConfig) {
+    return client.getInstance().getPDTopology(options)
+  }
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-op/src/apps/Metrics/context.ts:20]: For `Metrics` App, we don't need these APIs. I think we just need `metricsQueryGet`.

File: ui/packages/tidb-dashboard-for-op/src/apps/Metrics/context.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    return client.getInstance().getGrafanaTopology(options)
+  }
+
+  getAlertManagerTopology(options?: ReqConfig) {
+    return client.getInstance().getAlertManagerTopology(options)
+  }
+
+  getAlertManagerCounts(address: string, options?: ReqConfig) {
+    return client.getInstance().getAlertManagerCounts({ address }, options)
+  }
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-op/src/apps/Metrics/context.ts:50]: We don't need them.

File: ui/packages/tidb-dashboard-for-op/src/apps/Metrics/context.ts (hunk 1 above)
Comment (baurine) [inline ui/packages/tidb-dashboard-for-op/src/apps/Metrics/context.ts:57]: Yep, we don't need it either, I think. We just pass what we need.

File: ui/packages/tidb-dashboard-for-op/src/apps/Metrics/meta.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { AppstoreOutlined } from '@ant-design/icons'
+
+export default {
+  id: 'metrics',
+  routerPrefix: '/metrics',
+  icon: AppstoreOutlined,
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-op/src/apps/Metrics/meta.ts:6]: Should we use a different icon for it?

File: ui/packages/tidb-dashboard-for-dbaas/src/apps/Metrics/context.ts [hunk 5]
```diff
@@ -52,7 +52,7 @@ class DataSource implements IOverviewDataSource {
 
 const ds = new DataSource()
 
-export const ctx: () => IOverviewContext = () => ({
+export const ctx: () => IMetricsContext = () => ({
   ds,
   cfg: { apiPathBase: client.getBasePath() }
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-dbaas/src/apps/Metrics/context.ts:57]: If we don't use `apiPathBase` inside the app, we don't need it.

File: ui/packages/tidb-dashboard-lib/src/apps/Metrics/context/index.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    options?: ReqConfig
+  ): AxiosPromise<number>
+
+  metricsQueryGet(
+    endTimeSec?: number,
+    query?: string,
+    startTimeSec?: number,
+    stepSec?: number,
+    options?: ReqConfig
+  ): AxiosPromise<MetricsQueryResponse>
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/src/apps/Metrics/context/index.ts:42]: I think we just need this API.

File: ui/packages/tidb-dashboard-lib/src/apps/Metrics/data/MetricsItems.tsx [hunk 7]
```diff
@@ -1,9 +1,9 @@
 const MetricsItems = [
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/src/apps/Metrics/data/MetricsItems.tsx:1]: For non-component, we prefer to use camel naming style for variable and file, for example: `metricsItems`

**Comment by baurine** [comment]
image

Align the chart with the toolbar may be better.

File: ui/packages/tidb-dashboard-lib/src/components/MetricChart/seriesRenderer.tsx [hunk 8]
```diff
@@ -1,8 +1,8 @@
-import { BarSeries, LineSeries, ScaleType } from '@elastic/charts'
+import { BarSeries, LineSeries, ScaleType, AreaSeries } from '@elastic/charts'
 import { DataPoint } from '@lib/utils/prometheus'
 import React from 'react'
 
-export type GraphType = 'bar_stacked' | 'line'
+export type GraphType = 'bar_stacked' | 'line' | 'stack'
```
Comment (shhdgit) [inline ui/packages/tidb-dashboard-lib/src/components/MetricChart/seriesRenderer.tsx:5]:
```suggestion
export type GraphType = 'bar_stacked' | 'area_stack' | 'line'
```

File: ui/packages/tidb-dashboard-lib/src/components/MetricChart/seriesRenderer.tsx [hunk 9]
```diff
@@ -17,6 +17,8 @@ export function renderQueryData(type: GraphType, qd: QueryData) {
       return renderStackedBar(qd)
     case 'line':
       return renderLine(qd)
+    case 'stack':
```
Comment (shhdgit) [inline ui/packages/tidb-dashboard-lib/src/components/MetricChart/seriesRenderer.tsx:20]:
```suggestion
    case 'area_stack':
```

File: ui/packages/tidb-dashboard-lib/src/components/MetricChart/seriesRenderer.tsx [hunk 10]
```diff
@@ -60,3 +62,20 @@ function renderLine(qd: QueryData) {
     />
   )
 }
+
+function renderStack(qd: QueryData) {
```
Comment (shhdgit) [inline ui/packages/tidb-dashboard-lib/src/components/MetricChart/seriesRenderer.tsx:66]:
```suggestion
function renderAreaStack(qd: QueryData) {
```

**Review by shhdgit — APPROVED** [review]
The rest LGTM.

**Review by baurine — APPROVED** [review]
LGTM

**Pull request merged**